File: Arrays/Second_Largest/better_sol.cpp
#include<bits/stdc++.h>
using namespace std;

void secondLargest(int arr[], int n){

    int largest = INT_MIN;
    int second_largest = INT_MIN;

    for(int i=0; i<n; i++){
        if (arr[i] > largest){
            largest = arr[i];
        }
    }

    for(int i=0; i<n; i++){
        if (arr[i] > second_largest && arr[i] != largest){
            second_largest = arr[i];
        }
    }
    cout << "Second largest element is: " << second_largest << endl;
}

int main()
{
    // Initialize the array with elements
    int arr[] = {1, 2, 4, 6, 7, 5};
    
    // Calculate the size of the array
    int n = sizeof(arr) / sizeof(arr[0]);
    
    // Call the function to find and print the second largest element
    secondLargest(arr, n);

    return 0;
}

File: Arrays/Second_Largest/better_sol.py
def second_Largest(arr):

    largest = float('-inf')
    second_largest = float('-inf')

    for i in range(len(arr)):
        if arr[i] > largest:
            largest = arr[i]

    for i in range(len(arr)):
        if arr[i] > second_largest and arr[i] != largest:
            second_largest = arr[i]

    print("Largest element is:", largest)
    print("Second largest element is:", second_largest)

if __name__ == "__main__":
    arr = [10, 5, 8, 12, 15, 7]
    second_Largest(arr)